projects/plugins/super-cache/inc/delete-cache-button.php

```diff
@@ -108,6 +108,10 @@ function wpsc_admin_bar_delete_cache() {
 	$req_path = isset( $_POST['path'] ) ? sanitize_text_field( stripslashes( $_POST['path'] ) ) : '';
 	$valid_nonce = ( $req_path && isset( $_POST['nonce'] ) ) ? wp_verify_nonce( $_POST['nonce'], 'delete-cache-' . $_POST['path'] . '_' . $_POST['admin'] ) : false;
 
+	if ($valid_nonce) {
+		do_action('after_wpsc_delete_cache_directory', $req_path, $referer);
+	}
+
 	if ( $valid_nonce && $referer && $req_path && ( false !== stripos( $referer, $req_path ) || 0 === stripos( $referer, wp_login_url() ) ) ) {
```

Comment:
I assume we would only want to trigger this right before the redirect, if we have all the needed info? If so, we could place the hook inside the next conditional.

I also think we may want to rename the hook to be a bit more specific. Since it's outside of the `wpsc_delete_cache_directory` function and specific to the `wpsc_admin_bar_delete_cache`, we could mention that in the hook name, like so:

```suggestion
	if (
		$valid_nonce
		&& $referer
		&& $req_path
		&& (
			false !== stripos( $referer, $req_path )
			|| 0 === stripos( $referer, wp_login_url() )
		)
	) {
		/**
		 * Hook into the cache deletion process after a successful cache deletion from the admin bar button.
		 *
		 * @since $$next-version$$
		 *
		 * @param string $req_path Path of the page where the cache flush was requested.
		 * @param string $referer  Referer URL.
		 */
		do_action( 'wpsc_after_delete_cache_admin_bar', $req_path, $referer );

```

Reply:
Thanks for fast reply!
This is something I need to test on monday, I will let you know :)

Reply:
This is actually OK for my case :)